src/stores/LifecycleStore.ts

```diff
@@ -84,37 +86,46 @@ class LifecycleStore extends AsyncStore<IState> {
 }
 
 async function checkServerVersions(): Promise<void> {
-    const client = MatrixClientPeg.get();
-    if (!client) return;
-    for (const version of SUPPORTED_MATRIX_VERSIONS) {
-        // Check if the server supports this spec version. (`isVersionSupported` caches the response, so this loop will
-        // only make a single HTTP request).
-        // Note that although we do this on a reconnect, we cache the server's versions in memory
-        // indefinitely, so it will only ever trigger the toast on the first connection after a fresh
-        // restart of the client.
-        if (await client.isVersionSupported(version)) {
-            // we found a compatible spec version
-            return;
+    try {
+        const client = MatrixClientPeg.get();
+        if (!client) return;
+        for (const version of SUPPORTED_MATRIX_VERSIONS) {
+            // Check if the server supports this spec version. (`isVersionSupported` caches the response, so this loop will
+            // only make a single HTTP request).
+            // Note that although we do this on a reconnect, we cache the server's versions in memory
+            // indefinitely, so it will only ever trigger the toast on the first connection after a fresh
+            // restart of the client (which has just happened since it's started syncing, so this should
+            // never actually cause a request).)
```

Comment:
I don't think that's accurate. Or rather, it's accurate, but not for the reason given. The sync loop doesn't populate the `serverVersionsPromise` which is used by `isVersionSupported`.

Reply:
Ah, in my brief investigation it appeared to only do a single request, but not 100% sure so I've just removed that bit of the comment.